### PR TITLE
Remove SHIFT IN character bytes from Circuit ID

### PR DESCRIPTION
**Pull request by akshaynawale**
Some Vendors like Arista sends SHIFT IN (0x000f) prefix to the circuit id. This used to fail the circuit id parsing logic. This check will remove the 0x000f bytes prefix from the circuit id field if present. 
Added unit test for the same.
Signed-off-by: Akshay Navale <navale@fb.com>

```
go test
PASS
ok      _/home/navale/go/src/github.com/akshaynawale/dhcp/dhcpv4/ztpv4  0.004s
```

### CODE REVIEW

File: dhcpv4/ztpv4/parse_circuitid_test.go [hunk 1]
```diff
@@ -85,6 +86,7 @@ func TestParseCircuitID(t *testing.T) {
 		{name: "Cisco pattern", circuit: []byte("Gi1/10:2020"), want: &CircuitID{Slot: "1", Port: "10", Vlan: "2020"}},
 		{name: "Cisco Nexus pattern", circuit: []byte("Ethernet1/3"), want: &CircuitID{Slot: "1", Port: "3"}},
 		{name: "Juniper Bundle Pattern", circuit: []byte("ae52.0"), want: &CircuitID{Port: "52", SubPort: "0"}},
+		{name: "Arista Vlan pattern 1 with SHIFT IN", circuit: bytes.Join([][]byte{[]byte{0x00, 0x0f}, []byte("Ethernet14:Vlan2001")}, []byte{}), want: &CircuitID{Port: "14", Vlan: "Vlan2001"}},
```
Comment (pmazzini) [inline dhcpv4/ztpv4/parse_circuitid_test.go]: What is the `[]byte{}` for in the last parameter of bytes.Join?

**Comment by pmazzini** [comment]
Minor NIT. The rest looks good. Thank you @akshaynawale.

File: dhcpv4/ztpv4/parse_circuitid.go [hunk 2]
```diff
@@ -47,7 +48,13 @@ func ParseCircuitID(packet *dhcpv4.DHCPv4) (*CircuitID, error) {
 
 	// As per RFC 3046 sub-Option 1 is circuit-id. Look at 2.0 section in that RFC
 	// https://tools.ietf.org/html/rfc3046
-	circuitIdStr := string(relayOptions.Options.Get(dhcpv4.AgentCircuitIDSubOption))
+	cid := relayOptions.Options.Get(dhcpv4.AgentCircuitIDSubOption)
+	// Some Vendor like Arista sends SHIFT IN character i.e. 0x000f before circuitid
+	// remove it before matching against regexps.
+	if bytes.HasPrefix(cid, []byte{0x00, 0x0f}) {
```
Comment (insomniacslk) [inline dhcpv4/ztpv4/parse_circuitid.go]: you can simply do `cid = bytes.TrimPrefix(cid, []byte{0x00, 0x0f})` here, according to the documentation:
```
TrimPrefix returns s without the provided leading prefix string. If s doesn't start
with prefix, s is returned unchanged.
```

File: dhcpv4/ztpv4/parse_circuitid_test.go (hunk 1 above)
Comment (insomniacslk) [reply dhcpv4/ztpv4/parse_circuitid_test.go]: it's the separator used to join the two byte sequences - empty separator, basically it concatenates.

However I would recommend to change the whole call to `bytes.Join` to a simpler `[]byte("\x00'x0fEthernet14:Vlan2001")`

**Comment by insomniacslk** [comment]
@pmazzini please also have a last look. If you merge, please select squash and merge, so this becomes a single commit

**Pull request merged**